core/src/main/java/org/jruby/RubySignalException.java

```diff
@@ -0,0 +1,128 @@
+ *
+ * The contents of this file are subject to the Eclipse Public
+ * License Version 1.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of
+ * the License at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Software distributed under the License is distributed on an "AS
+ * IS" basis, WITHOUT WARRANTY OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * rights and limitations under the License.
+ *
+ * Alternatively, the contents of this file may be used under the terms of
+ * either of the GNU General Public License Version 2 or later (the "GPL"),
+ * or the GNU Lesser General Public License Version 2.1 or later (the "LGPL"),
+ * in which case the provisions of the GPL or the LGPL are applicable instead
+ * of those above. If you wish to allow use of your version of this file only
+ * under the terms of either the GPL or the LGPL, and not to allow others to
+ * use your version of this file under the terms of the EPL, indicate your
+ * decision by deleting the provisions above and replace them with the notice
+ * and other provisions required by the GPL or the LGPL. If you do not delete
+ * the provisions above, a recipient may use your version of this file under
+ * the terms of any one of the EPL, the GPL or the LGPL.
+ ***** END LICENSE BLOCK *****/
+
+package org.jruby;
+
+import static jnr.constants.platform.Signal.NSIG;
+
+import org.jruby.anno.JRubyClass;
+import org.jruby.anno.JRubyMethod;
+import org.jruby.runtime.ObjectAllocator;
+import org.jruby.runtime.Arity;
+import org.jruby.runtime.Block;
+import org.jruby.runtime.ThreadContext;
+import static org.jruby.runtime.Visibility.PRIVATE;
+import org.jruby.runtime.builtin.IRubyObject;
+import org.jruby.RubyBasicObject;
+import org.jruby.RubySignal;
+import org.jruby.util.TypeConverter;
+
+@JRubyClass(name="SignalException", parent="Exception")
+public class RubySignalException extends RubyException {
+    private static final ObjectAllocator SIGNAL_EXCEPTION_ALLOCATOR = new ObjectAllocator() {
+        @Override
+        public IRubyObject allocate(Ruby runtime, RubyClass klass) {
+            return new RubySignalException(runtime, klass);
+        }
+    };
+
+    protected RubySignalException(Ruby runtime, RubyClass exceptionClass) {
+        super(runtime, exceptionClass);
+    }
+
+    static RubyClass createSignalExceptionClass(Ruby runtime, RubyClass exceptionClass) {
+        RubyClass signalExceptionClass = runtime.defineClass("SignalException", exceptionClass, SIGNAL_EXCEPTION_ALLOCATOR);
+        signalExceptionClass.defineAnnotatedMethods(RubySignalException.class);
+
+        return signalExceptionClass;
+    }
+
+    @JRubyMethod(optional = 2, visibility = PRIVATE)
+    public IRubyObject initialize(IRubyObject[] args, Block block) {
+        final Ruby runtime = getRuntime();
+        int argnum = 1;
+        IRubyObject sig = runtime.getCurrentContext().nil;
```

Comment:
its probably better to receive `ThreadContext context` instead or simply do a `runtime.getNil()` if the context is not needed anywhere else.